src/lightning/pytorch/accelerators/cuda.py

```diff
@@ -73,29 +77,35 @@ def get_device_stats(self, device: _DEVICE) -> Dict[str, Any]:
         """
         return torch.cuda.memory_stats(device)
 
+    @override
     def teardown(self) -> None:
         _clear_cuda_memory()
 
     @staticmethod
+    @override
     def parse_devices(devices: Union[int, str, List[int]]) -> Optional[List[int]]:
         """Accelerator device parsing logic."""
         return _parse_gpu_ids(devices, include_cuda=True)
 
     @staticmethod
+    @override
     def get_parallel_devices(devices: List[int]) -> List[torch.device]:
         """Gets parallel devices for the Accelerator."""
         return [torch.device("cuda", i) for i in devices]
 
     @staticmethod
+    @override
     def auto_device_count() -> int:
         """Get the devices when set to auto."""
         return num_cuda_devices()
 
     @staticmethod
+    @override
     def is_available() -> bool:
         return num_cuda_devices() > 0
 
     @classmethod
+    @override
```

Comment:
Does the order of staticmethod/classmethod and override matter? In https://github.com/Lightning-AI/lightning/pull/18911/files#diff-0771b30c8f08ba845b153fb4c07cb219bb5648bed2abebdb95afd6def13b830eR881-R882 you used a different order

Reply:
Good catch; PEP-698 states 

> `@override` needs to execute before many special descriptor-based decorators like `@property`, `@staticmethod`, and `@classmethod`.

This means that `@override` should go _below_ those decorators.

However:

> The `@override` decorator needs to execute after ordinary decorators like `@functools.lru_cache` that use wrapper functions.

So `@override` should go _above_ these decorators.

In any case, it doesn't matter for any of the CI tests and PEP-698 says "we do not expect type checkers to validate the ordering of decorators".

Nevertheless, I will be consistent and modify the other PR. The ordering in this PR is the correct one.